src/satisfia/agents/makeMDPAgentSatisfia.py

```diff
@@ -0,0 +1,250 @@
+		assert allowNegativeCoeffs || lossCoeff4Time >= 0, "lossCoeff4time must be >= 0"
+		assert allowNegativeCoeffs || lossCoeff4DeltaVariation >= 0, "lossCoeff4DeltaVariation must be >= 0"
+		assert allowNegativeCoeffs || lossCoeff4Entropy >= 0, "lossCoeff4entropy must be >= 0"
+		assert allowNegativeCoeffs || lossCoeff4KLdiv >= 0, "lossCoeff4KLdiv must be >= 0"
+		assert allowNegativeCoeffs || lossCoeff4OtherLoss >= 0, "lossCoeff4OtherLoss must be >= 0"
+		assert lossCoeff4Entropy == 0 || lossCoeff4MP == 0 || (uninformedPolicy in self.params), "uninformedPolicy must be provided if lossCoeff4MP > 0 or lossCoeff4Entropy > 0"
+
+		if verbose or debug:
+			print("makeMDPAgentSatisfia with parameters", self.params);
+
+	def __getitem__(self, name):
+		return self.params[name]
+
+	"""The dependency/callback graph of the following functions is partially recursive 
+		and involves aggregation (MIN, MAX, E) operations as follows:
+  
+		simulate
+		→ localPolicy
+		  → aspiration4state
+			→ minAdmissibleV, maxAdmissibleV
+			  → MIN(minAdmissibleQ), MAX(maxAdmissibleQ)
+				→ E(minAdmissibleV), E(maxAdmissibleV) (RECURSION)
+		  → estAspiration4action
+			→ minAdmissibleV, maxAdmissibleV, minAdmissibleQ, maxAdmissibleQ
+		  → combinedLoss
+			→ Q, .., Q6, Q_DeltaSquare, Q_ones
+			  → propagateAspiration (see below)
+			  → E(V), ..., E(V6), E(V_DeltaSquare), E(V_ones)
+				→ localPolicy (RECURSION)
+				→ E(Q), ..., E(Q6), E(Q_DeltaSquare), E(Q_ones) (RECURSION)
+			→ otherLoss_action
+			  → propagateAspiration (see below)
+			  → E(otherLoss_state)
+				→ localPolicy (RECURSION)
+				→ E(otherLoss_action) (RECURSION)
+			→ similarly for other loss components (RECURSION)
+		→ expectedDelta, varianceOfDelta, transition
+		→ propagateAspiration
+		  → aspiration4state
+		→ simulate (RECURSION)"""
+
+	# Utility function for deriving transition probabilities from the transition function:
+	# Remark: later the following should actually be provided by the env/world:
+
+	@lru_cache(maxsize=None)
+	def transitionDistribution(self, state, action):
+		return distribution.infer(lambda: self.world.transition(state, action))
+
+	# Compute upper and lower admissibility bounds for Q and V that are allowed in view of maxLambda and minLambda:
+
+	# Compute the Q and V functions of the classical maximization problem (if maxLambda==1)
+	# or of the LRA-based problem (if maxLambda<1):
+
+	@lru_cache(maxsize=None)
+	def maxAdmissibleQ(self, state, action): # recursive
+		if verbose or debug:
+			print("maxAdmissibleQ, state", state, "action", action, "...")
+
+		# register (state, action) in global store (could be anywhere, but here is just as fine as anywhere else)
+		self.stateActionPairsSet.insert((state, action))
+
+		Edel = expectedDelta(state, action)
+		if state.terminateAfterAction:
+			q = Edel
+		else:
+			# Bellman equation
+			q = distribution.infer(lambda: self.maxAdmissibleV(self.world.transition(state, action))).E() # recursion
```

Comment:
We should use an instance of `MDPWorldModel` here (see branch `world_model`) and leave the calculation (or estimation) of expected values of functions of successor states (like Q in this case) to that instance, using its method `expectation`, https://github.com/pik-gane/satisfia/blob/world_model/src/world_model/world_model.py#L58 . Here, this would look like:
```
q = Edel + self.world_model.expectation(state, action, self.maxAdmissibleV) # recursion
```
(I've also added the missing `Edel` here)